### PR TITLE
[file_length_] Generic plugin to monitor the number of lines (or variation thereof) of arbitrary files

### PR DESCRIPTION
**Pull request by shtrom**
Signed-off-by: Olivier Mehani <shtrom@ssji.net>

### CODE REVIEW

**Comment by sumpfralle** [comment]
Thank you for contributing to this plugin!

Do you have an idea, *why* it could be useful to use something else, than `GAUGE` for the field type?

**Comment by shtrom** [comment]
Oh yeah, I just added this because I wanted to monitor error log files. In this case, the number of lines is a bit pointless, particularly with log rotation. However, the rate of increase, with `DERIVE` can be indicative of issues if there are spikes and periods of higher error rates than usual.

File: plugins/system/file_length_ [hunk 1]
```diff
@@ -15,6 +15,7 @@ Useful for things such as lists (white, black, user, ...).
     env.category DEFAULTS_TO_system
     env.title OPTIONAL_TITLE
     env.logarithmic 1
+    env.type DEFAULT_TO_GAUGE
```
Comment (steveschnepp) [inline plugins/system/file_length_]: Make it `GAUGE` instead of `DEFAULT_TO_GAUGE`

**Comment by steveschnepp** [comment]
>  However, the rate of increase, with `DERIVE` can be indicative of issues if there are spikes and periods of higher error rates than usual.

I'd even argue that _indeed_ `DERIVE` might be better default than `GAUGE`...

File: plugins/system/file_length_ (hunk 1 above)
Comment (steveschnepp) [reply plugins/system/file_length_]: As I said, `DERIVE` looks like a better default. But I don't really use the plugin, so I don't really vote for any.

**Comment by sumpfralle** [comment]
> I'd even argue that indeed DERIVE might be better default than GAUGE...

But only for error log files, I guess. There can be other uses of this plugin (watching the number of lines in an htpasswd file or similar).
How about adding a line of comment to the documentation, that mentions the use case for `DERIVE`?

**Comment by shtrom** [comment]
I think given the name and the fact that the plugin was already in, we should keep it as `GAUGE`, otherwise we risk breaking some use-cases relying on it being the default.

I'll add the comment (and a bugfix) a bit later on 👍

**Comment by sumpfralle** [comment]
That looks good to me - thank you!

**Pull request merged**